at-quannguyen/W3/dictionary.rb

```diff
@@ -0,0 +1,111 @@
+require 'time'
+# class dictionary
+class Dictionary
+  def initialize
+    @list_dictionary = {}
+    @id = 1
+  end
+
+  def add(dict, des)
+    if Dictionary.check_empty_by_vocab(@list_dictionary, dict)
+      @list_dictionary[@id] =
+        { dict: dict, des: des, lookup: 0, history_lookup: 0 }
+      @id += 1
+      return true
+    end
+    false
+  end
+
+  def lookup(vocab)
+    result = Dictionary.find_dict_by_vocab(@list_dictionary, vocab)
+    if result
+      sleep 0.5
+      result[1][:lookup] += 1
+      result[1][:history_lookup] = Time.new.strftime('%m/%d/%Y %H:%M:%S')
+      return result[1][:des]
+    end
+    nil
+  end
+
+  def size
+    @list_dictionary.length
+  end
+
+  def remove(vocab)
+    result = Dictionary.find_dict_by_vocab(@list_dictionary, vocab)
+    check = result ? result[0] : nil
+    @list_dictionary.delete(check) ? true : false
+  end
+
+  def pop
+    @list_dictionary.delete(@list_dictionary.keys.max)
+  end
+
+  def update(vocab, desc)
+    unless Dictionary.check_empty_by_vocab(@list_dictionary, vocab)
+      result = Dictionary.find_dict_by_vocab(@list_dictionary, vocab)
+      @list_dictionary.delete(result[0])
+      return add(vocab, desc)
+    end
+    false
+  end
+
+  def random
+    keys = @list_dictionary.keys
+    key_rand = keys[rand(keys.size)]
+    result = @list_dictionary[key_rand]
+    { result[:dict] => result[:des] }
+  end
+
+  def favorite
+    arr = []
+    result = nil
+    lookup = @list_dictionary.values.map { |e| e[:lookup] }
+    ck_favorite(arr, lookup, result)
+  end
+
+  def ck_favorite(arr, lookup, result)
+    unless lookup.max.zero?
+      @list_dictionary.values.each { |e| arr << e if e[:lookup] == lookup.max }
+      if arr.length > 1
+        nummin = arr.map { |e| Time.parse(e[:history_lookup]) }.min
```

Comment:
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/atruby-team/internship-18-summer/blob/6f22e2e17b7a320ef0dcf3e430a8fa8270c1e786/at-quannguyen/W3/dictionary.rb#L71"></span>Rails/TimeZone: Do not use <code>Time.parse</code> without zone. Use one of <code>Time.zone.parse</code>, <code>Time.current</code>, <code>Time.parse.in_time_zone</code>, <code>Time.parse.utc</code>, <code>Time.parse.getlocal</code>, <code>Time.parse.xmlschema</code>, <code>Time.parse.iso8601</code>, <code>Time.parse.jisx0301</code>, <code>Time.parse.rfc3339</code>, <code>Time.parse.httpdate</code>, <code>Time.parse.to_i</code>, <code>Time.parse.to_f</code> instead. (https://github.com/rubocop-hq/rails-style-guide#time, http://danilenko.org/2012/7/6/rails_timezones)</td>
    </tr>
  </tbody>
</table>